pmc-rs/src/counter.rs

```diff
@@ -360,9 +368,6 @@ fn init_pmc_once() -> Result<(), Error> {
             };
             return;
         }
-
-        // Register the signal handler
-        signal::watch_for(&[libc::SIGBUS, libc::SIGIO]);
```

Comment:
This signal handler is used in the [`read()` (and other) methods][read] to watch for the SIGBUS/SIGIO signals that the [libpmc docs] state:

> Applications using PMCs are required to handle the following signals:
> 
>      SIGBUS  When the hwpmc(4) module is unloaded using kldunload(8), pro-
>       cesses that have PMCs allocated to them will be sent a SIGBUS
>       signal.
> 
>      SIGIO   The hwpmc(4) driver will send a PMC owning process a SIGIO signal
>       if:
> 
>       o  If any process-mode PMC allocated by it loses all its target
>    		processes.
> 
>       o  If the driver encounters an error when writing log data to a
>   		configured log file.  This error may be retrieved by a subse-
>    		quent call to pmc_flush_logfile().

This follows the "Typical Program Flow" described in the same man page.

Your commit message for the change removing this mentions it was because you're seeing SIGIO during test execution, and that perhaps the test harness was the source of these; perhaps we disable SIGIO handling in tests if that's the case? What do you think?

[libpmc docs]:https://man.freebsd.org/cgi/man.cgi?query=pmc
[read]:https://github.com/domodwyer/pmc-rs/blob/master/pmc-rs/src/counter.rs#L286

Reply:
Unfortunately those signals can be caused by lots of reasons:
* SIGBUS is usually caused by an out-of-bounds memory access, and the chance that a program can recover is very low.  I don't think this library should try to catch this signal, because it could mask other errors in the program.
* SIGIO is normally sent by sockets with O_ASYNC set.  A pmc-rs consumer might legitimately want to use that signal, but it can't if the library registers its own signal handler.

The bottom line is that signal handlers need to be registered at the application level, not by a library, unless it's opt-in and explicit.  I don't think there's anything useful that the pmc-rs library can do with these signals that won't also potentially interfere with consuming applications.

Reply:
Makes sense, I agree it's probably better for the application to register the handlers - I can add some docs that suggest it.

Probably worth `rm`-ing the whole signal module in that case :+1: